aten/src/ATen/core/TensorImpl.cpp

```diff
@@ -26,10 +26,18 @@ TensorImpl::TensorImpl(TensorTypeId type_id, ScalarType scalar_type, Allocator *
   }
 }
 
-TensorImpl::TensorImpl(Storage&& storage, TensorTypeId type_id, bool is_variable)
-    : TensorImpl(std::move(storage), type_id, dataTypeToScalarType(storage.dtype()), is_variable) {}
-
-TensorImpl::TensorImpl(Storage&& storage, TensorTypeId type_id, ScalarType scalar_type, bool is_variable)
+TensorImpl::TensorImpl(Storage storage, TensorTypeId type_id, bool is_variable)
+    :storage_(std::move(storage)),
+    storage_offset_(0),
+    sizes_{0},
+    strides_{1},
+    is_contiguous_(true),
+    numel_(0),
+    type_id_(type_id),
+    scalar_type_(dataTypeToScalarType(storage_.dtype())),
+    is_variable_(is_variable) {}
```

Comment:
This doesn't seem shorter. Why have two nearly identical versions of the constructor? Seems like the second one is totally unnecessary. (I wouldn't have peeped up about this, but your diff made the code longer)

Reply:
This is needed because you shouldn't call std::move() in one argument and use the moved thing in another argument, which was how the old code did it.

Reply:
Suggest deleting the second constructor.